R/bin_utils.R

```diff
@@ -0,0 +1,88 @@
+#' Get Bins of Metadata
+#'
+#' Creates unique age/heights-bins for each record to speed up joining
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset that is the same as that in `derive_params_growth_age()`/`derive_params_growth_height()`
+#'
+#' @param param Character vector
+#'
+#'   Accepted values: "AGE"/"HEIGHT_LENGTH"
+#'
+#' @family internal
+#' @keywords internal
+#' @return list of breaks and labels for age/height bins
+get_bins <- function(meta_criteria, param) {
```

Comment:
@rossfarrugia @Fanny-Gautier this function grabs the metadata AGE/HEIGHT_LENGTH variable and finds the unique "bins" that created agnostic of bin width, e.g [0,1) [1,2) [2, 3) ... etc